web/src/app/schedules/ScheduleOverrideForm.js

```diff
@@ -38,163 +39,158 @@ const query = gql`
   }
 `
 
-const styles = theme => ({
+const useStyles = makeStyles({
   tzNote: {
     display: 'flex',
     alignItems: 'center',
   },
 })
 
-@connect(state => ({ zone: urlParamSelector(state)('tz', 'local') }))
-@withStyles(styles)
-export default class ScheduleOverrideForm extends React.PureComponent {
-  static propTypes = {
-    scheduleID: p.string.isRequired,
+export default function ScheduleOverrideForm(props) {
+  const { add, remove, errors = [], scheduleID, value, ...formProps } = props
 
-    value: p.shape({
-      addUserID: p.string.isRequired,
-      removeUserID: p.string.isRequired,
-      start: p.string.isRequired,
-      end: p.string.isRequired,
-    }).isRequired,
+  const classes = useStyles()
+  const params = useSelector(urlParamSelector)
+  const zone = params('tz', 'local')
+  const [userErrors, setUserErrors] = useState([])
```

Comment:
Does this need to be stateful or can we just replace `userErrors` with a function that uses `errors`?

It feels like we're duplicating data here and trying to keep it in sync.

Reply:
It's an interesting case because we have the standard `errors` variable from the mutation, but have to create the user conflict errors array based off the query if the mutation fails with a `userID` fieldName in question. I'll update the name though so it's more clear what it's being used for